gui/src/main/java/cern/jarrace/inspector/gui/rest/ContainerService.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * © Copyright 2016 CERN. This software is distributed under the terms of the Apache License Version 2.0, copied
+ * verbatim in the file “COPYING”. In applying this licence, CERN does not waive the privileges and immunities granted
+ * to it by virtue of its status as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package cern.jarrace.inspector.gui.rest;
+
+import cern.jarrace.commons.domain.AgentContainer;
+import retrofit2.Call;
+import retrofit2.http.GET;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+import java.util.List;
+
+/**
+ * A service that can access a remote endpoint with methods related to {@link AgentContainer}s.
+ */
+public interface ContainerService {
```

Comment:
I am not sure, if I understand the concepts correctly, however as far as I see it, all the paths (e.g. "container/list") are encoded twice: Once in the server and once in this client interface. Right?

If so, then I wonder if there is no better way to do it? Why cant you have an interface in a common package?
I think using retrofit makes sense when you have no handle on the server API. But in our case we are designing both, so there should be a single place to change!!!!!